lib/rspec/core/memoized_helpers.rb

```diff
@@ -144,6 +146,25 @@ def __init_memoized
                       end
       end
 
+      # @private
+      def enforce_value_expectation(matcher, method_name)
+        matcher_supports_value_expectations = begin
+                                                matcher.supports_value_expectations?
+                                              rescue
+                                                true
+                                              end
+        return if matcher_supports_value_expectations
+
+        RSpec.deprecate(
+          "#{method_name} #{RSpec::Support::ObjectFormatter.format(matcher)}",
+          :message =>
+            "The implicit block expectation syntax is deprecated, you should pass " \
+            "a block to `expect` to use the provided block expectation matcher " \
+            "(#{RSpec::Support::ObjectFormatter.format(matcher)}), " \
+            "or the matcher must implement `supports_value_expectations?`."
```

Comment:
This message might need to be tweaked to deprecate `should` entirely but its a good interim message 👍

Reply:
I'll take care of adding deprecation messages for what we'll end up removing in `4-0-dev` 👍